src/lib.rs

```diff
@@ -67,3 +67,33 @@ unsafe extern "C" fn clock_gettime(
 
     retval
 }
+
+#[no_mangle]
+unsafe extern "C" fn getrandom(
+    buf: *mut libc::c_void,
+    mut buflen: libc::size_t,
+    flags: libc::c_uint,
+) -> libc::ssize_t {
+    // TODO: is this needed? Maybe just `buflen = buflen.min(libc::ssize_t::MAX)` ?
+    buflen = buflen.min(0x1FFFFFF);
+
+    if flags != 0 {
+        // no flags are supported on 3DS
+        *__errno() = libc::EINVAL;
+        return -1;
+    }
+
+    let ret = ctru_sys::PS_GenerateRandomBytes(buf, buflen as libc::c_uint) as libc::ssize_t;
+    if ret < 0 {
+        // this is kind of a hack, but at least gives some visibility to the
+        // error code returned by PS_GenerateRandomBytes I guess? Another option
+        // might be to panic, which could use a payload of a specific error type
```

Comment:
Panicking across FFI boundaries is undefined behavior:
https://doc.rust-lang.org/nomicon/ffi.html#ffi-and-panics

Reply:
Ah crap, I forgot about that. Guess that's not an option then.